mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +347,95 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _load_custom_metric_artifact(
+    dataset_name, temp_dir, artifact_name, raw_artifact, custom_metric_index, custom_metric_name
+):
+    if isinstance(raw_artifact, EvaluationArtifact):
+        return raw_artifact
+
+    exception_header = _get_custom_metric_exception_header(custom_metric_index, custom_metric_name)
+
+    # Given local path, type inference through file extension
+    if isinstance(raw_artifact, str):
+        artifact_path = pathlib.Path(raw_artifact)
+        ext_to_artifact_map = {
+            ".png": ImageEvaluationArtifact,
+            ".jpg": ImageEvaluationArtifact,
+            ".jpeg": ImageEvaluationArtifact,
+            ".json": JsonEvaluationArtifact,
+            ".npy": NpyEvaluationArtifact,
+            ".csv": CsvEvaluationArtifact,
+            ".parquet": ParquetEvaluationArtifact,
+        }
+        if artifact_path.is_file() and artifact_path.suffix in ext_to_artifact_map.keys():
+            # copy file to temp location with file name in the defined format
+            artifact_file_name = _gen_log_key(artifact_name, dataset_name) + artifact_path.suffix
+            artifact_file_local_path = temp_dir.path(artifact_file_name)
+            shutil.copyfile(raw_artifact, artifact_file_local_path)
+
+            mlflow.log_artifact(artifact_file_local_path)
+            return ext_to_artifact_map[artifact_path.suffix](
+                uri=mlflow.get_artifact_uri(artifact_file_name)
+            )
+        raise MlflowException(
+            f"{exception_header} produced an unsupported artifact '{artifact_name}' with path "
+            f"'{raw_artifact}'. The supported file extensions are:"
+            f".png, .jpg, .jpeg, .json, .npy, .csv, .parquet"
+        )
+
+    import matplotlib.pyplot as pyplot
+
+    # Given object, type inference
+    type_to_ext_map = {
+        pd.DataFrame: ".csv",
+        np.ndarray: ".npy",
+        pyplot.Figure: ".png",
+    }
+
+    type_to_artifact_map = {
+        pd.DataFrame: CsvEvaluationArtifact,
+        np.ndarray: NpyEvaluationArtifact,
+        pyplot.Figure: ImageEvaluationArtifact,
+    }
+
+    artifact_file_name = (
+        _gen_log_key(artifact_name, dataset_name)
+        + f"{type_to_ext_map.get(type(raw_artifact), '.json')}"
```

Comment:
Did you mean a `_` before `artifact_file_name`? I used `.get` because in the event that the type of `raw_artifact` does not exist in `type_to_ext_map`, it will default to use the `.json` extension. I realized I don't really need the string formatting, it should be simplified to:
```python
artifact_file_name = _gen_log_key(artifact_name, dataset_name) + type_to_ext_map.get(type(raw_artifact), '.json')
```

Reply:
Did you mean a `_` before `artifact_file_name`? I used `.get` because in the event that the type of `raw_artifact` does not exist in `type_to_ext_map`, it will default to use the `.json` extension. I realized I don't really need the string formatting, it should be simplified to:
```python
artifact_file_name = _gen_log_key(artifact_name, dataset_name) + type_to_ext_map.get(type(raw_artifact), '.json')
```

Reply:
Nvm, looked through `gen_log_key`. We don't want to separate the generated name from the extension :)